datajunction-server/datajunction_server/sql/parsing/ast.py

```diff
@@ -2640,6 +2650,74 @@ async def compile(self, ctx: CompileContext):
         if self._is_compiled:
             return
 
+        def _compile(info: Tuple[Column, List[TableExpression]]):
+            """
+            Given a list of table sources, find a matching origin table for the column.
+            """
+            col, table_options = info
+            matching_origin_tables = 0
+            for option in table_options:
+                namespace = col.namespace[0].name if col.namespace else None
+                table_alias = option.alias.name if option.alias else None
+                if not namespace or (namespace and namespace == table_alias):
+                    result = option.add_reference_column(col)
+                    if result:
+                        matching_origin_tables += 1
+                        col._is_compiled = True
```

Comment:
If this internal flag is set because `result = option.add_reference_column(col)` was succesfull, then shouldn't we just set it as part of the `add_reference_column` function call?

Reply:
The problem is that it's not clear to me that all that's needed to compile `col` is calling `option.add_column_reference`. We're using it as a shortcut here, but I'm not sure that other calls that use this same function are using it this way?